src/presets/github-pages.ts

```diff
@@ -0,0 +1,18 @@
+import fsp from "node:fs/promises";
+import { join } from "pathe";
+import { defineNitroPreset } from "../preset";
+
+export const githubPages = defineNitroPreset({
+  extends: "static",
+  commands: {
+    deploy: "npx gh-pages -d ./public",
+  },
+  hooks: {
```

Comment:
Maybe we could add 404.html to prerender routes by default too?

Reply:
I guess we do it by default in Nuxt, but does it make sense for Nitro to do so when there might not be an HTML renderer?